src/components/screens/editor/editor-screen.tsx

```diff
@@ -182,6 +193,283 @@ const EditorScreen: React.FC<Props> = ({
+      selectedIntent: { ...selectedIntent, trainingMaterials: { ...selectedIntent.trainingMaterials, [key]: undefined } },
+      intents: intents.map(item => item.id === selectedIntent.id ? selectedIntent : item)
+    });
+    setEditingTrainingMaterial(false);
+    setEditedTrainingMaterial(undefined);
+  }
+
+  /**
+   * Event handler for save training material click
+   * 
+   * @param action action type: update / create
+   */
+  const onSaveTrainingMaterialClick = async (action: string) => {
+    let updatedTrainingMaterial: TrainingMaterial;
+    let updatedIntent: Intent;
+    if (!accessToken || !editedTrainingMaterial?.type || !trainingMaterial || !selectedIntent?.id || !intents) {
+      return;
+    }
+    const key = objectKeyConversion(editedTrainingMaterial.type);
+
+    if (action === "update" && editedTrainingMaterial.id) {
+      try {
+        updatedTrainingMaterial = await Api.getTrainingMaterialApi(accessToken).updateTrainingMaterial({
+          trainingMaterialId: editedTrainingMaterial.id,
+          trainingMaterial: editedTrainingMaterial
+        });
+        updatedIntent = await Api.getIntentsApi(accessToken).updateIntent({
+          intentId: selectedIntent.id,
+          intent: { ...selectedIntent, trainingMaterials: { 
+              ...selectedIntent.trainingMaterials,
+              [key]: editedTrainingMaterial.id
+            }
+          },
+          storyId: storyId
+        });
+      } catch (error) {
+        throw error;
+      }
+      setStoryData({
+        ...storyData,
+        trainingMaterial: trainingMaterial.map(item => item.id === updatedTrainingMaterial.id ? updatedTrainingMaterial : item),
+        selectedIntent: updatedIntent,
+        intents: intents.map(item => item.id === updatedIntent.id ? updatedIntent : item)
+      });
+    }
+    if (action === "create") {
+      try {
+        updatedTrainingMaterial = await Api.getTrainingMaterialApi(accessToken).createTrainingMaterial({
+          trainingMaterial: editedTrainingMaterial
+        });
+        updatedIntent = await Api.getIntentsApi(accessToken).updateIntent({
+          intentId: selectedIntent.id,
+          intent: { ...selectedIntent, trainingMaterials: {
+              ...selectedIntent.trainingMaterials,
+              [key]: editedTrainingMaterial.id
+            }
+          },
+          storyId: storyId
+        });
+      } catch (error) {
+        throw error;
+      }
+      setStoryData({
+        ...storyData,
+        trainingMaterial: [ ...trainingMaterial, updatedTrainingMaterial ],
+        selectedIntent: updatedIntent,
+        intents: intents.map(item => item.id === updatedIntent.id ? updatedIntent : item)
+      });
+    }
+    setEditingTrainingMaterial(false);
+    setSelectedTrainingMaterialType(null);
+  }
+
+  /**
+   * Event handler for updating knot info
+   * 
+   * @param knot knot with updated info
+   */
+  const onUpdateKnotInfo = async (event: React.ChangeEvent<any>, knot: Knot) => {
```

Comment:
Type change event and add tsdocs

Reply:
Typed change event, added tsdocs, removed excess params.